coala_quickstart/generation/SettingsClass.py

```diff
@@ -0,0 +1,256 @@
+        :param bear:
+            The current bear object.
+        :param trigger:
+            String of value either 'optional' or 'non-optional'
+            depending on type of settings
+        """
+        self.settings_bool = []
+        self.settings_others = []
+        self.fillup_settings(function, settings, bear, trigger)
+
+    def fillup_settings(self, function, settings, bear, trigger):
+        """
+        Fill settings_bool and settings_others depending upon whether the
+        particular setting by the bear takes a bool value or any other.
+        :param function:
+            The function object i.e. either the run() method or
+            the create_arguments() method of the bear.
+        :param settings:
+            Either a dict of non-optional settings of the form:
+            {'setting_name': ('Description.', <class 'type'>),}
+            or
+            a dict of optional settings of the form:
+            {'setting_name': default_values,}
+        :param bear:
+            The current bear object.
+        :param trigger:
+            String of value either 'optional' or 'non-optional'
+            depending on type of settings
+        """
+        for key in settings:
+            if trigger == 'optional':
+                self.fillup_optional_settings(key, function)
+            else:
+                self.fillup_non_optional_settings(key, function, bear)
+
+    def fillup_optional_settings(self, key, func):
+        """
+        Function to populate the optional settings
+        for the classes to store metadata.
+        :param key:
+            The setting value as a string.
+        :param func:
+            The function object. Either create_arguments() for linter bears
+            or run() for other bears.
+        """
+        present_in_annot = in_annot(func, key)
+        in_default_args(func, key)
+
+        if present_in_annot:
+            self.diff_bool_others(key, present_in_annot)
+        else:
+            self.diff_bool_others_default(key, get_default_args(func)[key])
+
+    def fillup_non_optional_settings(self, key, func, bear):
+        """
+        Function to populate the non-optional settings
+        for the classes to store metadata.
+        :param key:
+            The setting value as a string.
+        :param func:
+            The function object. Either create_arguments() for linter bears
+            or run() for other bears.
+        :param bear:
+            The bear object.
+        """
+        present_in_annot = in_annot(func, key)
+        present_in_default_args = in_default_args(func, key)
+        present_in_all_args = in_all_args(func, key)
+
+        if present_in_annot:
+            self.diff_bool_others(key, present_in_annot)
+        elif present_in_all_args and not present_in_default_args:
+            self.diff_bool_others(key, get_all_args(func)[key])
+        else:
+            self.parse_dep_tree(bear, key)
```

Comment:
why are we not doing this for optional settings?

Reply:
I don't think coala provides any way to provide the value for the optional settings, to `BEAR_DEPS`, please correct me if I am wrong.